apps/ogr2ogr_lib.cpp

```diff
@@ -4340,6 +4385,12 @@ SetupTargetLayer::Setup(OGRLayer *poSrcLayer, const char *pszNewLayerName,
                     poSrcFDefn->GetFieldDefn(iSrcField);
                 OGRFieldDefn oFieldDefn(poSrcFieldDefn);
 
+                if (iChangeWidthBy != 0 && oFieldDefn.GetType() == OFTReal &&
```

Comment:
this should be moved after the call to DoFieldTypeConversion() which might change the field type, unset the width, etc.